### PR TITLE
[PoC] Make to able to validate everything

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | https://github.com/yiisoft/validator/issues/101

### CODE REVIEW

File: src/DataSet/ScalarDataSet.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\DataSet;
+
+use Yiisoft\Validator\DataSetInterface;
+
+final class ScalarDataSet implements DataSetInterface
```
Comment (samdark) [inline src/DataSet/ScalarDataSet.php:9]: What is the use case for this one? It looks a bit out of place.

File: src/Rule/CompareTo.php [hunk 2]
```diff
@@ -129,10 +129,6 @@ protected function validateValue($value, ValidationContext $context = null): Res
     {
         $result = new Result();
 
-        if ($this->compareValue === null) {
```
Comment (samdark) [inline src/Rule/CompareTo.php:132]: Why?

Comment (xepozz) [reply src/Rule/CompareTo.php:132]: Because `null` is an allowable value too.

File: src/DataSet/ScalarDataSet.php (hunk 1 above)
Comment (xepozz) [reply src/DataSet/ScalarDataSet.php:9]: [It's used for strings, numbers and other scalar types](https://github.com/yiisoft/validator/pull/116/files/f26c7594b8275bb1002476dd6a5425f4628bba12#diff-9dad0f0dabc730596b3399e6fe25aed903dba8bc7836fe778fce387cf9f3e111R62)

**Review by samdark — APPROVED** [review]

**Pull request merged**